### DISCUSSION TITLE
How to load types from bunch of header files

### DISCUSSION THREAD

**Discussion by nitrogar** [Vector35/binaryninja-api#3617] [Q&A] [answered]
Basicly I have an open source library with /include/*.h folder. And I want to load all types and enums in one go while automatically resolve the dependencies between them.

Is there a way to do it with the new clang parser ? Or I am forced to add types one by one using the UI ?

**Comment by psifertex** [comment, marked as answer]
Sorry for the delay in replying -- there's a new action on the [latest dev](https://docs.binary.ninja/guide/index.html#updates) called `Import Header File` that's also under the `Analysis` menu that at least adds a UI for single files. 

That said, there's also https://api.binary.ninja/binaryninja.typeparser-module.html?highlight=parse_type#binaryninja.typeparser.TypeParser.parse_types_from_source that will take multiple files.

**Reply by psifertex** [reply]
Small update to note this is available in the menu on recent devs or the stable out in a few weeks:
![Screenshot 2023-01-10 at 5 39 50 PM](https://user-images.githubusercontent.com/140215/211677491-4a4d5f9c-efdb-4363-a572-c5f9a84e0b3b.png)
